opentreemap/treemap/views/misc.py

```diff
@@ -227,3 +229,23 @@ def instance_geojson(instance):
                  }))
 
     return [instance_geojson(instance) for instance in instances]
+
+
+def error_page(status_code):
+    template = '%s.html' % status_code
+
+    def inner_fn(request):
+        # API requests with an unhandled error should return JSON, not HTML
+        if ((request.path.startswith('/api/') or
+             'application/json' in request.META.get('HTTP_ACCEPT'))):
+            response = HttpResponse(json.dumps(
+                {'status': 'Failure', 'reason': 'An unhandled error occured'}),
```

Comment:
Saying there was an "unhanded error" in the 404 case is incorrect. I would suggest checking ``status_code`` to decide what the value of ``reason`` should be.